components/constants/pref_names.h

```diff
@@ -85,7 +85,8 @@ inline constexpr char kBraveShieldsSettingsVersion[] =
 inline constexpr char kDefaultBrowserPromptEnabled[] =
     "brave.default_browser_prompt_enabled";
 
-inline constexpr char kWebDiscoveryEnabled[] = "brave.web_discovery_enabled";
+inline constexpr char kWebDiscoveryExtensionEnabled[] =
```

Comment:
I'm a bit confused here, usages of `kWebDiscoveryExtensionEnabled` are guarded by extensions enabled (in some places and missing in others), but this is not so why are there guards for the pref itself?

Reply:
It seems there's no reason to have a guard around the `kWebDiscoveryExtensionEnabled` pref and the guards are also inconsistent and sometimes include the native pref so I think we should just drop them. Guards should generally be consistent between gn and cpp so if neither this file nor the pref declaration itself is guarded by enable extensions then access to it shouldn't be guarded by extensions enabled either

Reply:
i added the missing guards in two places. in general, if there is a guard around both the extension pref and the native pref, it's for pref migration

Reply:
but there's no reason for this added complexity and there are still missing guards. Also this pref should move out of components/constants and into web_discovery.

Reply:
added more guards. also decided to keep the pref here since it's shared between the extension and the native component. since the native component is behind a separate flag, it didn't seem like the best place for the shared pref.